### PR TITLE
added explanation of performance issues on osxfs from @dsheets post

### PR DESCRIPTION
**Pull request by londoncalling**
### Proposed changes

Added topic on "Performance issues, solutions, and roadmap". 
### Related issues

issue: https://github.com/docker/pinata/issues/5564

forum topic: https://forums.docker.com/t/file-access-in-mounted-volumes-extremely-slow-cpu-bound/
### Please take a look

@samoht @dsheets @justincormack @thaJeztah 

Signed-off-by: Victoria Bialas victoria.bialas@docker.com

### CODE REVIEW

File: docker-for-mac/osxfs.md [hunk 1]
```diff
@@ -152,3 +143,200 @@ Extended attributes are not yet supported.
...
+extremely
+slow](https://forums.docker.com/t/file-access-in-mounted-volumes-extremely-slow-cpu-bound/),
+this topic provides an explanation of the issues, what we are doing about it,
+how the community can help us, and what you can expect in the future.
+This explanation is a slightly re-worked version of an [understanding
+performance
+post](https://forums.docker.com/t/file-access-in-mounted-volumes-extremely-slow-cpu-bound/8076/158?u=orangesnap)
+from David Sheets (@dsheets) on the [Docker development
+team](https://forums.docker.com/groups/Docker) to the forum topic just
+mentioned. We wanted to surface it in the documentation for wider reach.
```
Comment (dsheets) [inline docker-for-mac/osxfs.md]: > We wanted to surface it in the documentation for wider reach.

reads awkwardly but is probably fine. The text is in the present with the reader.

File: docker-for-mac/osxfs.md [hunk 2]
```diff
@@ -152,3 +143,200 @@ Extended attributes are not yet supported.
...
+fact at run-time, we maintain coherence during its hundreds of thousands of file
+system accesses to manipulate temporary state. There is no "correct" solution in
+this case. Either ember needs to change, the volume mount needs to have
+coherence properties specified on it somehow, some heuristic needs to be
+introduced to detect this access pattern and compensate, or the behavior needs
+to be indicated via, e.g., extended attributes in the OS X file system.
+
+These two examples come from performance use cases contributed by users and they
+are incredibly helpful in prioritizing aspects of file system performance to
+improve. I am personally developing statistical file system trace analysis tools
```
Comment (dsheets) [inline docker-for-mac/osxfs.md]: s/I am personally/We are/

File: docker-for-mac/osxfs.md [hunk 3]
```diff
@@ -152,3 +143,200 @@ Extended attributes are not yet supported.
...
+to characterize slow-performing workloads more easily in order to decide what to
+work on next.
+
+Under development, we have:
+
+1. A Linux kernel module to reduce data path latency by 2/7 copies and 2/5
```
Comment (dsheets) [inline docker-for-mac/osxfs.md]: s/kernel module/kernel patch/

**Comment by londoncalling** [comment]
Great @dsheets , thanks.

**Comment by londoncalling** [comment]
@dsheets @thaJeztah @MagnusS @dave-tucker @mstanleyjones @johndmulhausen Can someone give this a final review and approve my changes so I can merge? Thanks!

**Comment by londoncalling** [comment]
@dsheets :-) how about now? merge if you think it's good, thanks!

**Pull request merged**